### PR TITLE
Fixing routing bug for callback

### LINKED ISSUE #670: Add callback result tests

### LINKED ISSUE #689: Wrong callback receiver
```
 [2019-03-08T05:22:18Z DEBUG node_runtime] Runtime error: wasm callback execution failed with error: Wasmer(CallError(Resolve(ExportNotFound { name: "near_func_promiseResolved" })))
[2019-03-08T05:22:18Z INFO  client] Producing block index: 206767, beacon hash = `9H6hHvLz8VtCaMEsRMjSjZtPsRFt2b5Dtugx58ZDq91M`, shard hash = `GGww65DL3ttx75Mmc8ZFwp3oVXUzi8bQxdSRxMivGeTR`, #tx=0, #receipts=1
[2019-03-08T05:22:18Z DEBUG client] Input Transactions: [ReceiptTransaction {
    originator: "studio-892vqj5ni",
    receiver: "studio-z4zgpixqt",
    nonce: `4XsmQoTFUo7V1d4LxuwDLP6hyiMiZ1JqUtmuoQTnLntL`,
    body: Callback(
        CallbackResult {
            info: CallbackInfo {
                id: `GxxsQC2PveTHxKMA6F58Uj8Xg6AwzBaWEwH8KzVUgjfi`,
                result_index: 0,
                receiver: studio-892vqj5ni
            },
            result: `{"result":"joinResult"}`
        }
    )
} -> TransactionResult {
    status: Failed,
    logs: [
        "Runtime error: wasm callback execution failed with error: Wasmer(CallError(Resolve(ExportNotFound { name: \"near_func_promiseResolved\" })))"
    ],
    receipts: [
        `3pznyUChNuHHiYPtroTFVrvFCD6ToTyjb6yMGw7FGzvC`
    ]
}
```
The bug is in this line: 
https://github.com/nearprotocol/nearcore/blob/master/node/runtime/src/lib.rs#L409

It should be `callback_info.receiver` instead.
Need to add tests for his.

### PR DESCRIPTION
**Pull request by evgenykuzyakov**
Adding integration tests for promises.
Have to upgrade `hello.wasm`. Updating encoder, decoder and near.ts.
Fixes #689 
Fixes #670

### CODE REVIEW

File: nearlib/test/integration.test.js [hunk 1]
```diff
@@ -241,6 +241,245 @@ describe('with deployed contract', () => {
     });
 });
 
+describe('with promises', () => {
+    let contract, contract1, contract2;
+    let oldLog;
+    let logs;
+    let contractName = 'test_' + Date.now();
+    let contractName1 = 'test_' + Math.random();
```
Comment (bowenwang1996) [inline nearlib/test/integration.test.js]: Should we use some seed?

Comment (evgenykuzyakov) [reply nearlib/test/integration.test.js]: It should be random enough to avoid collisions.

Comment (vgrichina) [reply nearlib/test/integration.test.js]: @evgenykuzyakov let's pull it into separate test module?

Comment (evgenykuzyakov) [reply nearlib/test/integration.test.js]: Ok. I'll do this when I add more promise tests (for errors and joined promises)

**Pull request merged**